hackoregon/team-budget
# config to keep database secrets separate from settings

# Background
With this new configuration approach, we can now accomplish two goals:
- test the Django app with queries directly through to the AWS-hosted PostgreSQL instance
- protect the database and Django secrets so they do not accidentally leak into GitHub or Docker Hub

With a little extra work, each developer should also be able to setup a local instance of PostgreSQL and swap in a copy of `project_config.py` that contains connection info for the local PostgreSQL.  [BE CAREFUL TO ADD ANY ADDITIONAL `project_config.py` DERIVATIVES AND COUSINS TO `.gitignore`]

# Testing
Here's one way to test this new secrets configuration model:

- Fork/clone/pull the latest code from *team-budget* repo
- `git checkout secrets`
- make a copy of the config template: `cp /budget_proj/project_config_template.py /budget_proj/project_config.py`
- edit `project_config.py` and insert the required data for all five variables
- make sure `django_rest_swagger` and `psycopg2` are in requirements.txt
- setup the virtual environment as before:
```
virtualenv -p python3 budget_venv
source budget_venv/bin/activate
pip install -r requirements.txt
```
- run `python3 budget_proj/manage.py runserver`
- browse to http://127.0.0.1:8000/ocrb/, and/or http://127.0.0.1:8000/summary/, and/or http://127.0.0.1:8000/kpm/ 

## Expected Result
This should successfully demonstrate that you're connecting to the AWS EC2-hosted instance of PostgreSQL and pulling data directly from the cloud through your locally-hosted Django application.  As long as you see JSON data from any of those endpoints, your connection to the database in AWS was successful.
